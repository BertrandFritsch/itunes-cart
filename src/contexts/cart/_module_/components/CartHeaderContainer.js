// @flow

import type { State } from '../flowTypes';
import type { Dispatch } from 'redux';

import { connect } from 'react-redux';
import { createSelector } from 'reselect';
import actions from '../actions';
import CartHeader from './CartHeader';

// use a memoized selector to re-compute the props only when the items in the cart have changed
const getCartItems = createSelector(
  (state: State) => state.cart.get('cartItems'),
  cartItems => ({
    itemCount: cartItems.count(),
    totalPrice: cartItems.reduce((s, item) => s + item.trackPrice, 0)
  })
);

export default connect(
  (state: State) => getCartItems(state),
  (dispatch: Dispatch<*>) => ({
    onCheckout: () => dispatch(actions.checkoutRequested())
  })
)(CartHeader);
