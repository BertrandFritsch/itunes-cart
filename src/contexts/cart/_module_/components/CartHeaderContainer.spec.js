import React from 'react';
import { Provider } from 'react-redux';
import { Map } from 'immutable';
import renderer from 'react-test-renderer';
import CartHeaderContainer from './CartHeaderContainer';

describe('CartHeaderContainer', () => {
  let store;

  beforeEach(() => {
    const state = {
      cart: Map({
        cartItems: Map({
          trackId: '1',
          trackName: 'track1',
          trackArtist: 'trackArtist',
          artworkUrl60: 'http://...',
          trackPrice: 0.99
        })
      })
    };
    store = ({
      default: () => {
      },
      subscribe: () => {
      },
      dispatch: jest.fn(),
      getState: () => state
    });
  });

  it('should render correctly', () => {
    const wrapper = renderer.create(
      <Provider store={ store }>
        <CartHeaderContainer />
      </Provider>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
