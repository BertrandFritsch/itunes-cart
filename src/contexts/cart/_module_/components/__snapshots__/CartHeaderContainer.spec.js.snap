// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartHeaderContainer should render correctly 1`] = `
<div
  className="cart-item"
>
  <span
    className="cart-header-item-count"
  >
    5 items - 
  </span>
  <span
    className="cart-header-total-price"
  >
    NaN $
  </span>
  <button
    className="cart-header-checkout"
    onClick={[Function]}
  >
    Checkout
  </button>
</div>
`;
